dlnd/dlnd_datamode_glc_forcing_mod.F90

```diff
@@ -113,85 +119,79 @@ subroutine dlnd_datamode_glc_forcing_init_pointers(exportState, sdat, dfields, m
+         if (ChkErr(rc,__LINE__,u_FILE_u)) return
 
-      ! The following maps stream input fields to export fields that have an ungridded dimension
-      call dshr_dfield_add(dfields, sdat, state_fld='Sl_tsrf_elev', strm_flds=strm_flds_tsrf, state=exportState, &
-           logunit=logunit, mainproc=mainproc, rc=rc)
-      if (ChkErr(rc,__LINE__,u_FILE_u)) return
-      call dshr_dfield_add(dfields, sdat, state_fld='Sl_topo_elev', strm_flds=strm_flds_topo, state=exportState, &
-           logunit=logunit, mainproc=mainproc, rc=rc)
-      if (ChkErr(rc,__LINE__,u_FILE_u)) return
-      call dshr_dfield_add(dfields, sdat, state_fld='Flgl_qice_elev', strm_flds=strm_flds_qice, state=exportState, &
-           logunit=logunit, mainproc=mainproc, rc=rc)
-      if (ChkErr(rc,__LINE__,u_FILE_u)) return
+         strm_fld = 'Sl_topo_elev'//trim(nec_str)
+         call shr_strdata_get_stream_pointer( sdat, trim(strm_fld), strm_Sl_topo_elev(ng)%strm_ptr, requirePointer=.true., &
+              errmsg=trim(subname)//'ERROR: '//trim(strm_fld)//' must be associated for dlnd glc_forcing datamode', rc=rc)
+         if (ChkErr(rc,__LINE__,u_FILE_u)) return
 
-      deallocate(strm_flds_tsrf)
-      deallocate(strm_flds_topo)
-      deallocate(strm_flds_qice)
+         strm_fld = 'Flgl_qice_elev'//trim(nec_str)
+         call shr_strdata_get_stream_pointer( sdat, trim(strm_fld), strm_Flgl_qice_elev(ng)%strm_ptr, requirePointer=.true., &
+              errmsg=trim(subname)//'ERROR: '//trim(strm_fld)//' must be associated for dlnd glc_forcing datamode', rc=rc)
+         if (ChkErr(rc,__LINE__,u_FILE_u)) return
+      end do
 
    end subroutine dlnd_datamode_glc_forcing_init_pointers
 
    !===============================================================================
-   subroutine dlnd_datamode_glc_forcing_advance(exportState, rc)
-
-      ! input/output variables
-      type(ESMF_State)      , intent(inout) :: exportState
-      integer               , intent(out)   :: rc
+   subroutine dlnd_datamode_glc_forcing_advance()
 
       ! local variables
-      integer           :: n
-      real(r8), pointer :: fldptr2(:,:)
-      character(len=*), parameter :: subname='(dlnd_datamode_glc_forcing_advance): '
+      integer :: ni,ng
+      Character(len=*), parameter :: subname='(dlnd_datamode_glc_forcing_advance): '
       !-------------------------------------------------------------------------------
 
-      rc = ESMF_SUCCESS
-
       ! Set special value over masked points
-      call dshr_state_getfldptr(exportState, 'Sl_tsrf_elev', fldptr2=fldptr2, rc=rc)
-      if (chkerr(rc,__LINE__,u_FILE_u)) return
-      do n = 1,size(fldptr2,dim=2)
-         if (lfrac(n) == 0._r8) fldptr2(:,n) = 1.e30_r8
-      end do
+      ! Note that the inner dimension is the elevation class
+
+      elev_class_loop: do ng = 1,glc_nec+1
+         do ni = 1,size(Sl_tsrf_elev,dim=2)
+            if (lfrac(ni) == 0._r8) then
+               Sl_tsrf_elev(ng,ni) = 1.e30_r8
```

Comment:
Why this particular number? Can it be a parameter (maybe `SHR_CONST_SPVAL`)?
This question also applies to the use of this naked constant below.

Reply:
done

Reply:
done